container-engine-lib/lib/backend_impls/kubernetes/kubernetes_kurtosis_backend/kubernetes_kurtosis_backend_api_container_functions.go

```diff
@@ -411,13 +413,44 @@ func (backend *KubernetesKurtosisBackend) CreateAPIContainer(
 		return nil, stacktrace.Propagate(err, "An error occurred getting container ports from the API container's private port specs")
 	}
 
+	volumeAttrs, err := enclaveAttributesProvider.ForEnclaveDataDirVolume()
+	if err != nil {
+		return nil, stacktrace.Propagate(err, "An error occurred creating the labels for enclave data dir volume")
+	}
+
+	volumeLabelsStrs := map[string]string{}
+	for key, value := range volumeAttrs.GetLabels() {
+		volumeLabelsStrs[key.GetString()] = value.GetString()
+	}
+	if _, err = backend.kubernetesManager.CreatePersistentVolumeClaim(ctx, enclaveNamespaceName, enclaveDataDirVolumeName, volumeLabelsStrs, enclaveDataDirVolumeSize); err != nil {
+		return nil, stacktrace.Propagate(err, "An error occurred creating the persistent volume claim for enclave data dir volume for enclave '%s'", enclaveDataDirVolumeName)
+	}
+	shouldDeleteVolumeClaim := true
+
+	defer func() {
+		if !shouldDeleteVolumeClaim {
+			return
+		}
+		if err := backend.kubernetesManager.RemovePersistentVolumeClaim(context.Background(), enclaveNamespaceName, enclaveDataDirVolumeName); err != nil {
+			logrus.Warnf(
+				"Creating pod didn't finish successfully - we tried removing the PVC %v but failed with error %v",
+				enclaveDataDirVolumeName,
+				err,
+			)
+			logrus.Warnf("You'll need to clean up volume claim '%v' manually!", enclaveDataDirVolumeName)
+		}
+	}()
+
 	apiContainerContainers, apiContainerVolumes, err := getApiContainerContainersAndVolumes(image, containerPorts, envVarsWithOwnIp, enclaveDataVolumeDirpath)
 	if err != nil {
 		return nil, stacktrace.Propagate(err, "An error occurred getting API containers and volumes")
 	}
 
 	apiContainerInitContainers := []apiv1.Container{}
 
+	// Data is always persistent we can always restart like Docker
+	apiContainerRestartPolicy := apiv1.RestartPolicyOnFailure
```

Comment:
@leoporoli Questions here -

This is very similar to what Docker does; and docker always has persistent volume for `/kurtosis-data` but on Kubernetes that adds time on DO/AWS; on k3s it seems like no time is added